src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1172,30 +1184,47 @@ namespace winrt::TerminalApp::implementation
         if (_tabView.TabItems().IndexOf(tabViewItem, tabIndexFromControl))
         {
             // If IndexOf returns true, we've actually got an index
-            _RemoveTabViewItemByIndex(tabIndexFromControl);
+            auto tab{ _tabs.GetAt(tabIndexFromControl) };
+            _RemoveTab(tab);
         }
     }
 
     // Method Description:
     // - Removes the tab (both TerminalControl and XAML)
     // Arguments:
-    // - tabIndex: the index of the tab to be removed
-    void TerminalPage::_RemoveTabViewItemByIndex(uint32_t tabIndex)
+    // - tab: the tab to remove
+    winrt::Windows::Foundation::IAsyncAction TerminalPage::_RemoveTab(winrt::TerminalApp::TabBase tab)
     {
+        // Removing the tab from the collection should destroy its control and disconnect its connection,
+        // but it doesn't always do so. The UI tree may still be holding the control and preventing its destruction.
+        if (tab.ReadOnly())
+        {
+            ContentDialogResult warningResult = co_await _ShowCloseReadOnlyDialog();
+
+            if (warningResult == ContentDialogResult::Primary)
```

Comment:
Wait the `Primary` button is the "Cancel" button? That's... unexpected? confusing?

Reply:
How do you suggest to address this? Will comment be enough? Carlos and I agree that the safer "default choice" should be Cancel.

Reply:
Oh yea that makes sense. I guess I was just confused looking at the pics. Probably leaving a comment is more than enough to make sense of this ☺️